tycho-indexer/src/extractor/dynamic_contract_indexer/euler/metadata_generator.rs

```diff
@@ -0,0 +1,253 @@
+use tycho_common::models::{blockchain::Block, protocol::ProtocolComponent, TxHash};
+
+use crate::extractor::dynamic_contract_indexer::component_metadata::{
+    MetadataError, MetadataRequest, MetadataRequestGenerator, MetadataRequestType, RpcTransport,
+};
+
+pub struct EulerMetadataGenerator {
+    rpc_url: String,
+}
+
+impl EulerMetadataGenerator {
+    pub fn new(rpc_url: String) -> Self {
+        Self { rpc_url }
+    }
+}
+
+impl MetadataRequestGenerator for EulerMetadataGenerator {
+    fn generate_requests(
+        &self,
+        component: &ProtocolComponent,
+        block: &Block,
+    ) -> Result<Vec<MetadataRequest>, MetadataError> {
+        let mut requests = vec![];
+
+        let balance_transport = RpcTransport::new(
+            self.rpc_url.clone(),
+            "eth_call".to_string(),
+            vec![serde_json::json!([
+                  {
+                    "data": "0x0902f1ac", // getReserves()
+                    "to": component.id
+                  },
+                  block.number.to_string() //TODO: should it be formatted as a hex string?
+                ]
+            )],
+        );
+        requests.push(MetadataRequest::new(
+            format!("euler_balance_{}", component.id),
+            component.id.clone(),
+            MetadataRequestType::ComponentBalance { token_addresses: component.tokens.clone() },
+            Box::new(balance_transport),
+        ));
+
+        let limits_transport_0to1 = RpcTransport::new(
+            self.rpc_url.clone(),
+            "eth_call".to_string(),
+            vec![serde_json::json!([
+              {
+                "data": format!("0xaaed87a3000000000000000000000000{}000000000000000000000000{}", component.tokens[0], component.tokens[1]),
+                "to": component.id
+              },
+              block.number.to_string()
+            ])],
+        );
+        requests.push(MetadataRequest::new(
+            format!("euler_limits_{}", component.id),
```

Comment:
request_id will most likely be used for deduplication, so we need to be able to differentiate this request and the next one.